language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/TypeCastNumericExpression.java

```diff
@@ -107,6 +126,26 @@ public List<CodeAction> getCodeActions(Diagnostic diagnostic, DiagBasedPositionD
                 CodeActionKind.QuickFix));
     }
 
+    private boolean isNumericType(TypeSymbol typeSymbol) {
+        TypeDescKind typeKind = typeSymbol.typeKind();
+        return typeKind.isIntegerType() || typeKind == TypeDescKind.DECIMAL || typeKind == TypeDescKind.FLOAT;
+    }
+
+    /**
+     * Checks if the provided type is a string type. Cover strings and string unions
+     *
+     * @param typeSymbol Type symbol
+     * @return True if the type or a member of the type is a string
+     */
+    private boolean isStringType(TypeSymbol typeSymbol) {
```

Comment:
Shall we change the name of the code action to TypeCastInBinaryExpressionNode as the code action handles string types as well

Reply:
+1